src/libsync/abstractnetworkjob.cpp

```diff
@@ -289,7 +281,11 @@ QString AbstractNetworkJob::errorStringParsingBody(QByteArray *body)
 
 AbstractNetworkJob::~AbstractNetworkJob()
 {
-    setReply(nullptr);
+    if (!_finished && !_aborted && !_timedout) {
+        qCCritical(lcNetworkJob) << "Deleting running job" << this << parent();
+    }
+    delete _reply;
```

Comment:
Might just want to use a `QPointer` to manage `_reply`.

Reply:
_reply is a QPointer.

Reply:
Sorry, I meant a `QSharedPointer`.

Reply:
We don't share this pointer and I'm not sure QSharedPointer resets when the hold QObject is destroyed.